src/main/java/hudson/plugins/analysis/core/BuildResult.java

```diff
@@ -342,8 +349,21 @@ private boolean hasReferenceBuild() {
      * @return the reference build.
      */
     @Exported
-    public AbstractBuild<?, ?> getReferenceBuild() {
-        return owner.getProject().getBuildByNumber(referenceBuild);
+    @WithBridgeMethods(value=AbstractBuild.class, adapterMethod="getReferenceAbstractBuild")
+    public Run<?, ?> getReferenceBuild() {
+        return owner.getParent().getBuildByNumber(referenceBuild);
+    }
+
+    /**
+     * Added for backward compatibility. It generates <pre>AbstractBuild getReferenceBuild()</pre> bytecode during the build
+     * process, so old implementations can use that signature.
+     * 
+     * @see {@link WithBridgeMethods}
+     */
+    @Restricted(NoExternalUse.class)
+    @Deprecated
```

Comment:
I removed ```@Restricted```. The deprecated one is ok from my point of view, because it prevents a developer to use this method internally (from the class itself).